modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -12,31 +12,50 @@
  * @license  Loris license
  * @link     https://github.com/aces/Loris-Trunk
  */
-if (isset($_POST['tab'])) {
-    $tab = $_POST['tab'];
-
-    $db   =& \Database::singleton();
-    $user =& \User::singleton();
-
-    if ($tab == "candidateInfo") {
-        editCandInfoFields($db, $user);
-    } else if ($tab == "probandInfo") {
-        editProbandInfoFields($db, $user);
-    } else if ($tab == "familyInfo") {
-        editFamilyInfoFields($db, $user);
-    } else if ($tab == "deleteFamilyMember") {
-        deleteFamilyMember($db, $user);
-    } else if ($tab == "participantStatus") {
-        editParticipantStatusFields($db, $user);
-    } else if ($tab == "consentStatus") {
-        editConsentStatusFields($db, $user);
-    } else {
-        header("HTTP/1.1 404 Not Found");
-        exit;
-    }
+
+$user = \User::singleton();
+if (!$user->hasPermission('candidate_parameter_edit')) {
+    header("HTTP/1.1 403 Forbidden");
+    exit;
 }
 
+$tab = $_POST['tab'] ?? '';
+if ($tab === '') {
```

Comment:
Hey @xlecours, why not do something similar to what you're doing in modules/candidate_parameters/ajax/getData.php? I think you could save 4 lines by removing the Bad Request check and just leave it for the default case in the switch statement. Otherwise shouldn't the Bad Request be in getData.php?

Reply:
@maltheism yes, good catch. I will add this to getData because there is a slight difference between missing a field (bad request) and having a value mapping to an nonexistent function (not found).